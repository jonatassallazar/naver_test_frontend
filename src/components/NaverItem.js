import React, { useState, useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { TrashIcon, EditIcon } from './Icon';
import API from '../axios/instance';
import NaverSelected from './NaverSelected';
import store from '../store/store';

const NaverProfile = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  padding: 0 0 ${({ theme }) => theme.spacing.medium} 0;

  button {
    border: inherit;
    margin: 0;
    padding: 0;
    background: inherit;
    cursor: pointer;
    text-align: inherit;
    font-size: inherit;
    font-weight: inherit;
    color: inherit;
    font-family: inherit;
  }
`;

NaverProfile.Img = styled.img`
  width: 100%;
`;

NaverProfile.Name = styled.h3`
  font-weight: 600;
  font-size: 1.6rem;
  line-height: 1.8rem;
  margin: ${({ theme }) => theme.spacing.small} 0 0 0;
  text-decoration: none;
  color: ${({ theme }) => theme.colors.primary};
`;

NaverProfile.Role = styled.p`
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: ${({ theme }) => theme.spacing.tiny} 0 0 0;
  text-decoration: none;
  color: ${({ theme }) => theme.colors.primary};
`;

NaverProfile.Icons = styled.div`
  margin-top: ${({ theme }) => theme.spacing.small};

  button {
    border: none;
    margin: 0;
    padding: 0;
    background: none;
  }

  svg {
    margin-right: ${({ theme }) => theme.spacing.small};
    color: ${({ theme }) => theme.colors.primary};
    cursor: pointer;
  }
`;

const NaverItem = ({ id, photo, name, role }) => {
  const [showNaverModalSelected, setShowNaverModalSelected] = useState(false);
  const [naverSelected, setNaverSelected] = useState('');
  const { dispatch } = useContext(store);

  const handleDelete = () => {
    dispatch({ type: 'CONFIRMATION_DELETE_NAVER', id, dispatch});
  };

  const handleNaverShow = () => {
    API({
      url: `/navers/${id}`,
      method: 'get',
    })
      .then((res) => {
        setNaverSelected(res.data);
        setShowNaverModalSelected(true);
        console.log(naverSelected);
      })
      .catch((err) => console.log(err));
  };

  useEffect(() => {
    return () => {
      setShowNaverModalSelected(false);
    };
  }, []);

  return (
    <>
      {showNaverModalSelected ? (
        <NaverSelected
          naverSelected={naverSelected}
          setShowNaverModalSelected={setShowNaverModalSelected}
        />
      ) : undefined}
      <NaverProfile>
        <button onClick={handleNaverShow}>
          <NaverProfile.Img alt={name} src={photo} />
          <NaverProfile.Name>{name}</NaverProfile.Name>
          <NaverProfile.Role>{role}</NaverProfile.Role>
        </button>
        <NaverProfile.Icons>
          <button onClick={handleDelete}>
            <TrashIcon />
          </button>
          <Link to={`/navers/edit/${id}`}>
            <EditIcon />
          </Link>
        </NaverProfile.Icons>
      </NaverProfile>
    </>
  );
};

export default NaverItem;
