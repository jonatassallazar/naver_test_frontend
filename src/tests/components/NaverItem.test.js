import NaverItem from '../../components/NaverItem';
import customRender from './customRender';
import navers from '../fixtures/fixtures';
import 'jest-styled-components';

test('should match NaverItem snapshot', () => {

  const { container } = customRender(
    <NaverItem
      id={navers[1].id}
      photo={navers[1].url}
      name={navers[1].name}
      role={navers[1].job_role}
    />,
  );

  expect(container).toMatchSnapshot();
});
