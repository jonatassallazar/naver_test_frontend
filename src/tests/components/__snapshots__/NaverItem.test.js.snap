// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match NaverItem snapshot 1`] = `
.c5 {
  width: 1.8rem;
  height: 1.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  padding: 0 0 2.4rem 0;
}

.c0 button {
  border: inherit;
  margin: 0;
  padding: 0;
  background: inherit;
  cursor: pointer;
  text-align: inherit;
  font-size: inherit;
  font-weight: inherit;
  color: inherit;
  font-family: inherit;
}

.c1 {
  width: 100%;
}

.c2 {
  font-weight: 600;
  font-size: 1.6rem;
  line-height: 1.8rem;
  margin: 1.4rem 0 0 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #212121;
}

.c3 {
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 0.4rem 0 0 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #212121;
}

.c4 {
  margin-top: 1.4rem;
}

.c4 button {
  border: none;
  margin: 0;
  padding: 0;
  background: none;
}

.c4 svg {
  margin-right: 1.4rem;
  color: #212121;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <button>
      <img
        alt="Tester 2"
        class="c1"
        src="https://lorem.picsum.com"
      />
      <h3
        class="c2"
      >
        Tester 2
      </h3>
      <p
        class="c3"
      >
        DevOps
      </p>
    </button>
    <div
      class="c4"
    >
      <button>
        <svg
          class=" c5"
          version="1.1"
          viewBox="0 0 14 18"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M1 18H13V4H1V18ZM14 1H10.5L9.5 0H4.5L3.5 1H0V3H14V1Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <a
        href="/navers/edit/2"
      >
        <svg
          class=" c5"
          version="1.1"
          viewBox="0 0 19 19"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M0 15.25V19H3.75L14.81 7.94L11.06 4.19L0 15.25ZM18.41 4.34L14.66 0.589996L12.13 3.13L15.88 6.88L18.41 4.34Z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
